src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneState = [];
+
+  let stateCopy = {
+    ...state,
+  };
+
+  const addAction = 'addProperties';
+  const removeAction = 'removeProperties';
+  const clearAction = 'clear';
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case addAction:
+        stateCopy = {
+          ...stateCopy, ...extraData,
+        };
+        break;
+
+      case removeAction:
+        stateCopy = {
+          ...stateCopy,
+        };
```

Comment:
you already have the same destructive construction an the top (line 12) so there is no need to duplicate

Reply:
But construction at line 12 creates a copy of the state variable and on line 29 I copy the changed object to push in Clones state array. It's just not gonna work without this statement.